proposals/Proposal.md

```diff
@@ -0,0 +1,113 @@
+#﻿Improving reproducibility in science by adding provenance tracking to the EcoData Retriever.
+
+
+#Personal Details:-
+Name:- Prabh Simran Singh Baweja
+3rd Year, B.Tech in Computer Science and Engineering, IIIT Hyderabad, India.
+Email Id:- prabhsimransingh.baweja@gmail.com
+Website:- web.iiit.ac.in/~prabhsimransingh.baweja
+Phone Number:- +91-8790616675.
+
+
+#Abstract:
+The EcoData Retriever is a Python based tool for automatically downloading, cleaning up, and restructuring ecological data. Most of the steps are either related to data are either done manually, or using one-off scripts. Due to this, the process is not reproducible. The EcoData Retriever does most of this work, but is not able to keep track of the work that has been completed and therefore fails to support full reproducible workflows.
+
+
+#Technical Details:
+The project would be divided in three parts :-
+
+
+#1. Deciding on the usage of Provenance library or building our own functionality:-
+We will look at various open source provenance libraries, and check if any one of them can be integrated with the EcoData Retriever. If this fails, then we will have to implement the whole thing from scratch using a suitable database. This part of the project would set the environment for the future activities to take place.
+
+
+#2. Implementation of the first part:-
+At this stage, we would have a clear picture of what needs to be implemented. 
+
+#A).  
+If we are able to find a provenance library that can be integrated, then it’s various features need to be taken care of while integration. We can use various indexing techniques to store the pointers to the files that have been downloaded. The file pointers can be sorted according to their names, and then the pointers can point to the original files. This will reduce the time of search when we are check if a new file has already been downloaded or not. 
+
+
+#B.)     
+The second option is to implement it from scratch. In that case, we can design an algorithm which checks if the new file has already been downloaded. If not, this algorithm enables the storage of pointers to the file accordingly. We can use the above indexing techniques while implementing this part as well. 
+

```

Comment:
I would help to add some ideas about what information will be stored when recording the provenance of the data and the manipulations conducted by the retriever.